dataprofiler/profilers/numerical_column_stats.py

```diff
@@ -211,6 +211,12 @@ def _add_helper(
         :param other2: profile2 being added to self
         :return: None
         """
+        if not (
+            isinstance(other1, NumericStatsMixin)
+            and isinstance(other2, NumericStatsMixin)
+        ):
+            raise ValueError("Parameters must be of type NumericStatsMixin.")
+
```

Comment:
@junholee6a

Reply:
Removing this if-statement causes the mypy errors in the attached image, because the if-statement implicitly casts `other1` and `other2` to type `NumericStatsMixin`.

Like the other subclasses we discussed above, some of the methods in `NumericStatsMixin` override superclass methods despite not all input types being valid, which in principle violates LSP. I was thinking of fixing this in the same way as the other subclasses in the next PR.

<img width="943" alt="Screenshot 2023-05-17 at 2 17 10 PM" src="https://github.com/capitalone/DataProfiler/assets/53921230/0a33e252-f239-4c6a-9523-ec07da32178e">

Reply:
got it -- just make sure to bullet point out the fixes in the next PR in the issue too https://github.com/capitalone/DataProfiler/issues/747